components/layout/fragment.rs

```diff
@@ -921,7 +922,34 @@ impl Fragment {
             specific: specific,
             inline_context: None,
             pseudo: node.get_pseudo_element_type().strip(),
-            flags: FragmentFlags::empty(),
+            flags: FragmentFlags::with_quirks_mode(node.quirks_mode()),
+            debug_id: DebugId::new(),
+            stacking_context_id: StackingContextId::new(0),
+        }
+    }
+
+    /// Constructs a new `Fragment` from a node, overriding its style
+    pub fn from_node_and_style<N: ThreadSafeLayoutNode>(node: &N,
+                                                        style: Arc<ServoComputedValues>,
+                                                        specific: SpecificFragmentInfo)
+                                                        -> Fragment {
+        let writing_mode = style.writing_mode;
+
+        let mut restyle_damage = node.restyle_damage();
+        restyle_damage.remove(RECONSTRUCT_FLOW);
```

Comment:
If I read prperly, this single line is usually what is different where I changed the calls from `from_opaque_node_...` to this constructor. I don't know what are the consequences.